### ISSUE TITLE
Parse.User.logIn from Cloud Code fails

### ISSUE THREAD

**Issue by dnyle** [ParsePlatform/Parse-SDK-JS#189]
Hi,

I implemented a custom login function in cloud code which was working fine but as I tried it out yesterday I get "invalid login parameters" and I'm definitely sending the right credentials. On my way down to find the issue I even tried the most simple approach in a Parse.Cloud.define("user_login", .. block i called:

Parse.User.logIn(request.params.username, request.params.password, {
            success: function(user) {
                // Get user profile
                callback.success(user)
            },
            error: function(user, error) {
                callback.error(error);
            }
        })

but it still leaves me with the error. 

I saw there was a change in the SDK 3 days ago maybe it's somehow connected

Btw I'm not really sure if the JavaScript and the Cloud functionality are connected in Parse, please feel free to update me ;)

Regards Daniel

**Comment by dnyle** [comment]
FYI:

When I call the function from the API Console with Run as ... current user with valid session and send the user credentials I get:
{
    "code": 141,
    "error": "undefined"
}

**Comment by andrewimm** [comment]
Which version of the SDK is your cloud code using? You can find out by viewing your app's logs, and looking at the header of the page, or by running `parse jssdk` in your app's directory.

**Comment by dnyle** [comment]
E2016-01-27T17:11:00.941Z]v1168 Ran cloud function user_login with:
  Input: {"password":"...","username":"..."}
  Result: invalid login parameters
E2016-01-27T17:11:37.678Z]v1168 Ran cloud function user_login for user ... with:
  Input: {"password":"...","username":"..."}
  Result: undefined

v1168

Thanks

**Comment by andrewimm** [comment]
Sorry, I meant the version of the SDK, not the current deployment. In the header of the logs page, it tells you when the last deploy was, and what JS SDK version is used.

**Comment by dnyle** [comment]
Latest deploy: v1175 - Wednesday January 27th 2016 at 7:30:09pm (+0100) - Parse JavaScript SDK Version: 1.6.14

**Comment by andrewimm** [comment]
Cool, then we can not worry about the changes to login that went out with 1.7.0.

If you can provide a small, complete test case that reproduces this error reliably, we might be able to help. Otherwise, we're in the dark, as it's probably not tied to the SDK, but we'll do our best.

**Comment by dnyle** [comment]
Ok will do asap,

thanks in advance

**Comment by dnyle** [comment]
EDIT: I edit on top cause it's good new ;)

I just had the idea creating another app and upload the CloudCode for my user login there. Actually it worked so maybe my DB/App seems kind of corrupted. Might consider clean up everything and give it another try.

// Actual message 
Actually I don't know how u normally conduct your tests but as far as it concerns my case following approach fails 100% and that's the minimal setup possible:

I created a user: which works fine doing it with the API Console, in the data browser or a custom register function I created in cloud code do validate parameters and execute further steps on signUp.

Anyway after creating the user, for simplicity I used username="username", password="password", I call the cloud function user_login (from API Console or from my App)  passing username and password but to simplify it even more and eliminate any errors that might be occur on the transmission, I hardcoded the values into the cloud function:

```
Parse.Cloud.define("user_login", function(request, response) {
                   Parse.User.logIn("username", "password", {
                                    success: function(user) {
                                    esponse.error(user)
                                    },
                                    error: function(object, error) {
                                    response.error(error.message)
                                    }
                                    })
                   })
```

```
RESPONSE
{
    "code": 141,
    "error": "invalid login parameters"
}
```

I discovered the problem after I switched to an implementation using Promises, but even restoring the above code the result was the same. Been working on other stuff lately but the code was defiantly working before as I used it quite often to run my app

Anyway I gonna call the Parse.User.LogIn method directly now and might do my other setup in the afterSave method which seems to work. 

Thanks for your efforts, from my side no critical issue ;) but keep me posted if something comes up

Regards Daniel
